### PR TITLE
Remove check for older versions 2

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
New version of #122 

Started with removing old version in OwncloudVersion and then changing the checks to true, simplifying whereever possible.

To test, please use nextcloud/android#2178

Signed-off-by: tobiasKaminsky tobias@kaminsky.me

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
Again, shouldn't we go for support of N >= 10 since we have gotten recent reports with people working with Nc 10? While it is EOL and people should upgrade anyways I am unsure if breaking the client(s) is the right way to go.

**Comment by tobiasKaminsky** [comment]
NC10 support is done via #127. This also allows oC to work.

**Comment by AndyScherzinger** [comment]
:+1: Nice :)

**Comment by mario** [comment]
👍

**Comment by tobiasKaminsky** [comment]
1.0.39 will be in next 3.0.2 and as this changes quite a lot, I would like to have this first tested on dev version and then to be included in 1.0.40.

**Comment by AndyScherzinger** [comment]
> 1.0.39 will be in next 3.0.2 and as this changes quite a lot, I would like to have this first tested on dev version and then to be included in 1.0.40.

fine by me!

File: src/com/owncloud/android/lib/resources/users/GetRemoteUserInfoOperation.java [hunk 1]
```diff
@@ -131,9 +131,7 @@ protected RemoteOperationResult run(OwnCloudClient client) {
                 if (userInfo.getQuota() == null) {
                     userInfo.setQuota(new Quota());
                     userInfo.getQuota().setQuota(QUOTA_LIMIT_INFO_NOT_AVAILABLE);
-                }
-                
-                if (userInfo.getQuota().getQuota() == 0) {
+                } else if (userInfo.getQuota().getQuota() == 0) {
```
Comment (AndyScherzinger) [inline src/com/owncloud/android/lib/resources/users/GetRemoteUserInfoOperation.java]: @tobiasKaminsky please review, this is how I resolved the merge conflict while rebasing and this is imho better than "a second simple if by handling it in a else if"

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky can imho be merged if you are fine with my change after the rebase: https://github.com/nextcloud/android-library/pull/124#discussion_r173871472

File: src/com/owncloud/android/lib/resources/users/GetRemoteUserInfoOperation.java (hunk 1 above)
Comment (tobiasKaminsky) [reply src/com/owncloud/android/lib/resources/users/GetRemoteUserInfoOperation.java]: :+1:

**Comment by tobiasKaminsky** [comment]
Changes are fine :+1:

**Comment by mario** [comment]
@tobiasKaminsky @AndyScherzinger please check your mail before we decide to merge (or NOT merge) this.

**Comment by tobiasKaminsky** [comment]
Delaying it a bit until we have decision -> removing milestone

**Comment by mario** [comment]
So what do we do with this?

**Comment by AndyScherzinger** [comment]
@mario @tobiasKaminsky personally I think, postpone to 3.2. Then I think it can be merged since the app is broken of anything <= Nc 11.0.8. If there is a way to fic this + oC support I still think we should then support it + plus for Nc10 I already opened an issue on the client tracker: https://github.com/nextcloud/android/issues/2355 to display a pop-up right at the beginning when detecting an incompatible server.

**Comment by tobiasKaminsky** [comment]
Postpone to 3.2 and due to our little team and that we have that much of work, I really would like to drop oC / older server support.
If anyone of the community wants to give it a try, I am not strictly against it.
But our main goal is to support our own (and current) server versions.

**Comment by AndyScherzinger** [comment]
:+1:

**Comment by tobiasKaminsky** [comment]
Rebased, to get app PR working

**Comment by AndyScherzinger** [comment]
So merge then? @tobiasKaminsky @mario ?

**Comment by tobiasKaminsky** [comment]
:+1:

**Comment by AndyScherzinger** [comment]
we have a lot of multiple-time +1's while pullapprove seems to not having picked them up... :tada:

**Pull request merged**